cmd/juju/model/destroy.go

```diff
@@ -156,19 +181,78 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 	}
 	defer api.Close()
 
+	// Attempt to connect to the controller API.  If we can't, fail the destroy.
+	controllerAPI, err := c.getControllerAPI()
```

Comment:
Ok, I've added a common.ModelStatus API which is embedded in the Controller and ModelManager facades